src/pages/api/multipleProductValidity.ts

```diff
@@ -63,7 +96,10 @@ export default (req: NextApiRequestWithSession, res: NextApiResponse): void => {
         const rawProducts: Product[] = multiProductAttribute.products;
         const products: Product[] = rawProducts.map((rawProduct, i) => addErrorsIfInvalid(req, rawProduct, i));
 
-        updateSessionAttribute(req, MULTIPLE_PRODUCT_ATTRIBUTE, { products });
+        updateSessionAttribute(req, MULTIPLE_PRODUCT_ATTRIBUTE, {
+            products,
+            endTimesList: (req.body.listofEndTimes && req.body.listOfEndTimes.split(',')) || [],
```

Comment:
Do we need to do the removing of excess whitespace here? As this data is passed through to the matching JSON later on.

Reply:
On that, I think we're missing the changes in the userData file to add this endTimesList to the matching JSON.

Reply:
If you are referring to the split this is so that we can convert the object into an array before being added to the session. When it comes to the api it has to store it in a hidden input and therefore converts it into a string.

Reply:
Yeah, just thinking if we dont remove excess whitespace here, it will go into the session with whitespace.